MovieTime/MovieTime/Core/Router/MTRouter.swift

```diff
@@ -0,0 +1,65 @@
+//
+//  MTRouter.swift
+//  MovieTime
+//
+//  Created by Enrique Melgarejo on 10/01/18.
+//  Copyright © 2018 Choynowski. All rights reserved.
+//
+
+import UIKit
+
+enum MTAppState {
+    case splashScreen
+    case feed
+}
+
+final class MTRouter: NSObject {
+
+    static let shared = MTRouter()
+
+    private var currentState = MTAppState.splashScreen
+
+    private func changeRootViewController(forIdentifier identifier: String, storyboardName: String) {
+
+        guard let appDelegate = UIApplication.shared.delegate,
+            let window = appDelegate.window else { return }
+
+        let storyboard = UIStoryboard.init(name: storyboardName, bundle: nil)
+        let desiredViewController = storyboard.instantiateViewController(withIdentifier: identifier)
+
+        if let snapshot: UIView = (window?.snapshotView(afterScreenUpdates: true)) {
```

Comment:
I think you don't need the parentheses after `: UIView`, and a `guard let` would fit better here since this is already in the end of the scope, allowing you to reduce indentation